packages/core/src/RenderPipeline/BasicRenderPipeline.ts

```diff
@@ -236,6 +239,19 @@ export class BasicRenderPipeline {
     shaderPasses: ReadonlyArray<ShaderPass>,
     renderStates: ReadonlyArray<RenderState>
   ): void {
+    const { component, material } = subRenderElement;
+    // Union render global macro and material self macro
+    ShaderMacroCollection.unionCollection(
+      component._globalShaderMacro,
+      material.shaderData._macroCollection,
+      Shader._compileMacros
+    );
```

Comment:
duplicate union macro in `rendererqueue`

Reply:
done